usage.go

```diff
@@ -88,6 +88,13 @@ func (p *Parser) writeUsageForCommand(w io.Writer, cmd *command) {
 			fmt.Fprint(w, spec.placeholder)
 		}
 	}
+
+	// if the program supports subcommands and the command-line doesn't contain any,
+	// give a hint to the user about the existence of these subcommands.
+	if len(p.cmd.subcommands) > 0 && p.cmd == cmd {
```

Comment:
I think it would be better to drop the `p.cmd == cmd` condition here and print this extra string whenever subcommands are permitted. This would mean that when writing `./program --help`, the user would see that subcommands are supported. It would also work better with nested subcommands.

And the first condition should then change to `if len(cmd.subcommands) > 0` since this would cover the case where we are printing help text for a subcommand that has further nested subcommands.

Reply:
Done.